### PR TITLE
:running: Update linter and lint fixes

### PR DESCRIPTION
**Pull request by mcristina422**
This updates golangci-lint, moves the configuration all to `.golangci.yml` and then fixes the linter errors

### CODE REVIEW

File: pkg/internal/testing/integration/etcd.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (e *Etcd) setProcessState() error {
 		return err
 	}
 
-	e.processState.StartMessage = internal.GetEtcdStartMessage(e.processState.URL)
+	e.processState.HealthCheckEndpoint = "/health"
```
Comment (DirectXMan12) [inline pkg/internal/testing/integration/etcd.go]: o_O

File: pkg/manager/signals/signal_test.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ var _ = Describe("runtime signal", func() {
 			task := &Task{
 				ticker: time.NewTicker(time.Second * 2),
 			}
-			c := make(chan os.Signal)
+			c := make(chan os.Signal, 1)
```
Comment (DirectXMan12) [inline pkg/manager/signals/signal_test.go]: context on this one?

File: pkg/internal/testing/integration/etcd.go (hunk 1 above)
Comment (mcristina422) [reply pkg/internal/testing/integration/etcd.go]: `StartMessage` is deprecated. So staticcheck is failing. I was iffy on this change too, we can add `//nolint` to revert this change

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/testing/integration/internal/process.go#L40

File: pkg/manager/signals/signal_test.go (hunk 2 above)
Comment (mcristina422) [reply pkg/manager/signals/signal_test.go]: `SA1017: the channel used with signal.Notify should be buffered`
https://staticcheck.io/docs/checks#SA1017

File: .golangci.yml [hunk 3]
```diff
@@ -3,3 +3,5 @@ linters-settings:
     line-length: 170
   dupl:
     threshold: 400
+run:
+  deadline: 5m
```
Comment (kensipe) [inline .golangci.yml]: `deadline` is deprecated... we should switch to `timeout` 

details: https://github.com/golangci/golangci-lint/pull/822

File: hack/verify.sh [hunk 4]
```diff
@@ -24,29 +24,6 @@ go vet ${MOD_OPT} ./...
...
-    --enable=lll \
-    --enable=dupl \
-    --enable=goimports \
-    ./pkg/... ./examples/... .
-
-# TODO: Enable these as we fix them to make them pass
-#    --enable=gosec \
-#    --enable=maligned \
-#    --enable=safesql \
+golangci-lint run ./pkg/... ./examples/... .
```
Comment (kensipe) [inline hack/verify.sh]: The following linters have been removed and are NOT enabled by default

```
--enable=misspell \	   
   --enable=golint \	
   --enable=goconst \	
   --enable=unparam \	
   --enable=nakedret \	
   --enable=interfacer \	
   --enable=gocyclo \	
   --enable=lll \	
   --enable=dupl \	
   --enable=goimports \
```

Is this intended?   If we plan to keep them... it would be great to add them to the `.golangci.yml` file.

https://github.com/golangci/golangci-lint#enabled-by-default-linters

Comment (mcristina422) [reply hack/verify.sh]: No, this was not my intention. I misinterpreted how dropping the `--disable-all` flag worked. I'll make sure to explicitly enable the same ones in the config file

**Comment by mcristina422** [comment]
@kensipe I've reverted the etcd change. Since `staticcheck` isn't enabled currently this will pass linting

In a follow up we can debate enabling more like `staticcheck`, `stylecheck`, `whitespace` and others found in the more strict configuration that golangci uses themselves https://github.com/golangci/golangci-lint/blob/master/.golangci.yml

**Comment by kensipe** [comment]
@DirectXMan12 could we get a 2nd look on this?

File: pkg/manager/signals/signal_test.go (hunk 2 above)
Comment (gerred) [reply pkg/manager/signals/signal_test.go]: ah, prevents missing signals, ok.

File: hack/check-everything.sh [hunk 5]
```diff
@@ -74,7 +74,7 @@ function fetch_go_tools {
   header_text "Checking for gometalinter.v2"
   if ! is_installed golangci-lint; then
     header_text "Installing golangci-lint"
-    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
+    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.23.1
```
Comment (gerred) [inline hack/check-everything.sh]: long term maybe we introduce a tools.go to manage these dependencies with go modules - a few other tools do this.

**Comment by kensipe** [comment]
@mcristina422 looks like it needs a rebase... lets see if we can get some attention on this after the rebase... thx

File: pkg/client/fake/client_test.go [hunk 6]
```diff
@@ -235,7 +236,7 @@ var _ = Describe("Fake client", func() {
 					Namespace: "ns2",
 				}
 				obj := &corev1.ConfigMap{}
-				err = cl.Get(nil, namespacedName, obj)
```
Comment (camilamacedo86) [inline pkg/client/fake/client_test.go]: Why this change is required?

File: pkg/controller/controller_integration_test.go [hunk 7]
```diff
@@ -152,7 +152,7 @@ var _ = Describe("controller", func() {
 			By("Invoking Reconciling for an OwnedObject when it is updated")
 			newReplicaset := replicaset.DeepCopy()
 			newReplicaset.Labels = map[string]string{"foo": "bar"}
-			newReplicaset, err = clientset.AppsV1().ReplicaSets("default").Update(newReplicaset)
+			_, err = clientset.AppsV1().ReplicaSets("default").Update(newReplicaset)
```
Comment (camilamacedo86) [inline pkg/controller/controller_integration_test.go]: 👍 cool

File: hack/verify.sh [hunk 8]
```diff
@@ -24,29 +24,6 @@ go vet ${MOD_OPT} ./...
 
 header_text "running golangci-lint"
 
-golangci-lint run --disable-all \
```
Comment (camilamacedo86) [inline hack/verify.sh]: 👍 cool

File: .golangci.yml [hunk 9]
```diff
@@ -1,5 +1,27 @@
...
+    - goimports
+
 linters-settings:
   lll:
     line-length: 170
   dupl:
     threshold: 400
+
+run:
+  timeout: 5m
```
Comment (camilamacedo86) [inline .golangci.yml]: 👍 cool

File: .golangci.yml [hunk 10]
```diff
@@ -1,5 +1,27 @@
+linters:
+  disable-all: true
```
Comment (vincepri) [inline .golangci.yml]: I'd suggest to flip these. Enable all linters and explicitly disable the others, so we know which ones we are disabling and folks can pick up the task of removing a disabled linter from this list if they wanted to.

File: pkg/client/fake/client_test.go (hunk 6 above)
Comment (mcristina422) [reply pkg/client/fake/client_test.go]: When running the linter with staticcheck enabled (not enforced by this PR) it fails with

`SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (staticcheck)`

https://staticcheck.io/docs/checks#SA1012

File: .golangci.yml (hunk 10 above)
Comment (mcristina422) [reply .golangci.yml]: This is explicitly called out here because enable-all is deprecated  

https://github.com/golangci/golangci-lint/blob/b9eef79121fff235d0d794c176ffa2b3d9bd422f/.golangci.yml#L59-L60

Comment (vincepri) [reply .golangci.yml]: Oh well, it seems a bit counter intuitive to me, but nothing major

File: pkg/internal/testing/integration/etcd.go (hunk 1 above)
Comment (camilamacedo86) [reply pkg/internal/testing/integration/etcd.go]: Before it was `internal.GetEtcdStartMessage(e.processState.URL)`, however,, now it is receiving a fixed value. Is it accurate? Could not it cause issues?
